src/Pact/PersistPactDb.hs

```diff
@@ -282,7 +290,13 @@ writeUser s wt tn rk row = runMVState s $ do
 {-# INLINE writeUser #-}
 
 record :: (AsString k, PactDbValue v) => TxTable -> k -> v -> MVState p ()
-record tt k v = txRecord %= M.insertWith (flip (++)) tt [TxLog (asString (tableId tt)) (asString k) (toJSON v)]
+record tt k v = modify'
+    $ over txRecord
+    $ M.insertWith append tt [TxLog (asString (tableId tt)) (asString k) (toJSON v)]
+  where
+    -- strict append (it would be better to use a datastructure with efficient append)
+    append [] b = b
+    append (h:t) b = let !x = append t b in h : x
```

Comment:
Is this the right order? `M.insertWith (flip (++))` would get you `oldValue ++ newValue` per the [examples](https://hackage.haskell.org/package/containers-0.4.0.0/docs/src/Data-Map.html#insertWith). Your fold here looks like it's it will build `insertWith (++)`.

Reply:
Yeah, I just implemented a strict `++` and forgot to keep the `flip`